walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java

```diff
@@ -66,6 +66,42 @@
       availableTimes.add(TimeRange.fromStartEnd(startPointer, TimeRange.END_OF_DAY, true));
     }
 
-    return availableTimes;
+    Collection<TimeRange> optionalAvailableTimes = new ArrayList<>();
+    if (request.getOptionalAttendees().size() > 0) {
+      // get available times for optional attendees
+      MeetingRequest requestOptionalTimes = new MeetingRequest(request.getOptionalAttendees(), request.getDuration());
+      Collection<TimeRange> optionalTimes = query(events, requestOptionalTimes);
+
+      // find timeranges where available times for mandatory and optional attendees overlap
+      int j = 0;
+      for (TimeRange optionalTime: optionalTimes) {
+        for (int i = j; i < availableTimes.size(); i++) {
+          j = i;
```

Comment:
If I read it correctly, the size of availableTimes is never changed during these loops, thus j is always size() - 1 after the loop for the 1st optional time. Do you intend to keep comparing only to the last available time for the optional times starting from the 2nd optional time on?